### ISSUE TITLE
How to reproduce the DNA PPO fine-tuning

### ISSUE THREAD

**Issue by zhan8855** [ChenyuWang-Monica/DRAKES#1]
Congratulations for your awesome work!

I've successfully reproduced the protein PPO fine-tuning following your instructions, however, my results on DNA are not good. I would appreciate it if you could tell me more details on how to do PPO fine-tuning on DNA. 

Thank you very much in advance!

![Image](https://github.com/user-attachments/assets/17f76da2-6751-465c-956c-e922d7cd4f7d)

**Comment by ChenyuWang-Monica** [comment]
Thank you for the question. As a clarification, our method is direct reward backpropagation rather than PPO. Can you share more details how you do the fine-tuning on the DNA data? You should be able to reproduce the results by following the `finetune_reward_bp.py` file (with all the hyperparams we used set as default).

**Comment by zhan8855** [comment]
Thank you so much for your prompt reply! Yes I followed almost all the default config in ```finetune_reward_bp.py```, except for the batch size and accumulation steps. I halved batch size to fit in a single 40GB GPU and doubled gradient accumulation steps. After fine-tuning, I selected the last-step checkpoint for evaluation.

The wandb record of the run is [https://wandb.ai/zhan8855/reward_bp_final/runs/il4vpb0g](https://wandb.ai/zhan8855/reward_bp_final/runs/il4vpb0g). I would greatly appreciate it if you could help me check the configuration at  [https://wandb.ai/zhan8855/reward_bp_final/runs/il4vpb0g/overview](https://wandb.ai/zhan8855/reward_bp_final/runs/il4vpb0g/overview) and the training dynamics at [https://wandb.ai/zhan8855/reward_bp_final/runs/il4vpb0g/logs](https://wandb.ai/zhan8855/reward_bp_final/runs/il4vpb0g/logs). Do they match your results? Thank you very much in advance.

**Comment by ChenyuWang-Monica** [comment]
`gumbel_temp` needs to be set as 0.5.

**Comment by zhan8855** [comment]
Thank you so much!! Let me try it!

**Comment by zhan8855** [comment]
I have tried `gumbel_temp=0.5` and now I can match `Pred-Activity`, `ATAC-Acc`, and `App-Log-Lik` reported in the paper. However, the k-mer correlation is still not so good. I would greatly appreciate it if you could offer me more guidance. The new wandb record is [https://wandb.ai/zhan8855/reward_bp_final/runs/j36546di](https://wandb.ai/zhan8855/reward_bp_final/runs/j36546di) with configuration at [https://wandb.ai/zhan8855/reward_bp_final/runs/j36546di/overview](https://wandb.ai/zhan8855/reward_bp_final/runs/j36546di/overview) and training dynamics at [https://wandb.ai/zhan8855/reward_bp_final/runs/j36546di/logs](https://wandb.ai/zhan8855/reward_bp_final/runs/j36546di/logs). Do they match your results? Thank you very much again in advance.

![Image](https://github.com/user-attachments/assets/5b336b8d-c57c-4a42-848b-8d9d3db3ee33)
